### PR TITLE
CP DF ALS changes for CP LT

### PR DESCRIPTION
**Pull request by kmp5VT**
Allow one to access the CP factors optimized before the CP DF ALS computation and allow the the user to input an epsilon for these initial component factorizations

### CODE REVIEW

File: btas/generic/cp_df_als.h [hunk 1]
```diff
@@ -253,7 +253,7 @@ namespace btas {
     /// else if calculate_epsilon = true, returns 2-norm error between exact and approximate tensor
     /// else return -1
     double compute_comp_init(ind_t rank, ConvClass converge_test, size_t max_als = 1e4, bool fast_pI = true,
-                            bool calculate_epsilon = false, bool direct = true) {
+                            bool calculate_epsilon = false, bool direct = true, double cp3_precision = 1e-2) {
```
Comment (evaleev) [inline btas/generic/cp_df_als.h:256]: need dox

File: btas/generic/cp_df_als.h [hunk 2]
```diff
@@ -262,25 +262,26 @@ namespace btas {
...
         fit.set_norm(nrm(tensor_ref_right));
         CP_ALS<Tensor, FitCheck<Tensor>> CP3(tensor_ref_right);
         CP3.compute_rank_random(rank, fit, 100, true);
-        auto Ar = CP3.get_factor_matrices();
-        auto cur_dim = Ar.size();
+        init_factors_right = CP3.get_factor_matrices();
+        auto cur_dim = init_factors_right.size();
         for(size_t i = 1; i < cur_dim; ++i){
-          A.push_back(Ar[i]);
+          A.push_back(init_factors_right[i]);
```
Comment (evaleev) [inline btas/generic/cp_df_als.h:284]: can move `init_factors_right[I]`?

**Review by evaleev — CHANGES_REQUESTED** [review]

**Comment by evaleev** [comment]
@kmp5VT is this ready?

**Comment by kmp5VT** [comment]
> @kmp5VT is this ready?
I am fixing your requested changes and then it will be ready

**Comment by kmp5VT** [comment]
@evaleev This PR is ready to be pushed and integrated into TA

**Pull request merged**